### ISSUE TITLE
Please provide a direct way to assign an enum to object typed property in xaml

### ISSUE THREAD

**Issue by huoyaoyuan** [microsoft/microsoft-ui-xaml#3338]
# Discussion: Please provide a predictable way to assign an enum to object typed field

Assigning an enum value to *enum-typed* property is always simple, just `SomeProp="EnumMember"`. However, assigning it to untyped properties(typically `Content` or `DataContext`) has been very complicated.

1. `{x:Static ns:EnumType.EnumMember}`
The way in WPF. However, `x:Static` isn't available in UWP or WinUI xaml.

2. `{x:Bind ns:EnumType.EnumMember, Mode=OneTime}`
Works perfectly, *but only in context that `x:Bind` is available*. Cannot be used in ControlTemplate AFAIK.

3. `<ns:EnumType>EnumMember</ns:EnumType>`
This is institutive, **but it assigns the underlying integer**, not the enum. Cannot be used in scenarios that type must be correct.
The behavior is the same in UWP, so it isn't a regression.

4. `{MyEnumExtension Member=EnumMember}`
I tried this, but cannot make it work in WinUI 3.

## Proposed approach
I'd prefer way 3. However, It will be a breaking behavior change.

**Comment by ranjeshj** [comment]
@huoyaoyuan can you provide details on the scenario you are trying to solve ?

**Comment by huoyaoyuan** [comment]
> can you provide details on the scenario you are trying to solve ?

There are two root scenarios:
- Selecting **localizable** enum values in selector control.
- Displaying enum values as something localizable.

I'd make a more detailed requests for what I imagine the solution should be, but many approaches I've tried will require creating enum values and arrays in xaml. So I opened this request.

**Comment by stevenbrix** [comment]
Thanks @huoyaoyuan for filing this. It sounds like this is a feature request?

**Comment by huoyaoyuan** [comment]
@stevenbrix Yes. I'd expect to prepare a major update on xaml after you open source it.
